airflow-core/src/airflow/dag_processing/bundles/base.py

```diff
@@ -428,12 +429,23 @@ def acquire(self):
         self._update_version_file()
         if TYPE_CHECKING:
             assert self.lock_file_path
+        # Raises FileNotFoundError if the path doesn't exist,
+        # or PermissionError if access is denied
         self.lock_file = open(self.lock_file_path)
-        flock(self.lock_file, LOCK_SH)
+        # https://docs.python.org/3/library/fcntl.html#fcntl.flock
+        try:
+            flock(self.lock_file, LOCK_SH)
+        except (BlockingIOError, OSError) as e:
+            self._log_exc("Error acquiring lock")
+            raise AirflowFileLockAcquireException(self.lock_file_path, e) from e
```

Comment:
Do we really need these custom exceptions here? AirflowException isn't "special" in anyway, and in the past we (the project) have over used it, so I'm wondering if we need this at all?

Reply:
While I initially added them to distinguish file lock-related errors more clearly, I’m happy to simplify the code by removing the custom exceptions and using AirflowException instead.

Reply:
I think even `AirflowException` is wrong (as it makes everything the same type) -- why can't we let the existing OSError etc exceptions be caught here? i.e. even using AirflowException here is wrong I think.

With the addition of the `self._log_exc` you've added, wouldn't that be enough, and then letting the exception just bubble up might be enough?

Or another option, is to add a note to it https://peps.python.org/pep-0678/ -- this works on Python 3.11+, which is now most of the versions. I think I prefer this (and on 3.10 the behaviour is just nothing else)

So, on 3.10: log something about it, on 3.11 _don't_ log at site (else the exception stack gets logged twice, once there and once at the top level catch) and we use add_note to add information about what the cause was.

Reply:
> Or another option, is to add a note to it https://peps.python.org/pep-0678/ -- this works on Python 3.11+, which is now most of the versions. I think I prefer this (and on 3.10 the behaviour is just nothing else)

I test on python 3.10.12. I got an Attribute Error

<img width="1004" height="256" alt="image" src="https://github.com/user-attachments/assets/cfc47586-e5aa-4f5f-a924-d77fed71ac79" />

I'll refere to this code. Thanks 

```python
if hasattr(BaseException, "add_note"):
    # Py 3.11+
    raise_on_4xx_5xx = raise_on_4xx_5xx_with_note
```